testsuite/MDAnalysisTests/analysis/test_diffusionmap.py

```diff
@@ -69,6 +69,30 @@ def test_dist_weights(u):
                                 [.707, -.707, 0, 0]]), 2)
 
 
+def test_distvalues_ag_universe(u):
+    dist_universe = diffusionmap.DistanceMatrix(u, select='backbone').run()
+    ag = u.select_atoms('backbone')
+    dist_ag = diffusionmap.DistanceMatrix(ag).run()
+    assert_allclose(dist_universe.results.dist_matrix,
+                    dist_ag.results.dist_matrix)
+
+
+def test_distvalues_updating_ag(u):
+    selection = 'resname ALA'
+    ag = u.select_atoms(selection, updating=True)
```

Comment:
I'm not really understanding how this checks that an updating selection works, maybe I'm missing something obvious though.

Since universes are static, there shouldn't be fewer or more atoms that match "resname ALA" within the lifetime of a trajectory. Instead of you should aim to analyse something dynamic, like "ALA residues within a given distance of residue N".

Reply:
So the point of passing and UpdatingAtomGroup through and checking that everything still works as expected is because most Analysis methods in MDAnalysis use static defined numpy arrays. So if you end up with more or fewer entries than you'd expect, things can go wrong. Having a test that we can check works as expected is very important.

Reply:
I have a doubt here. Say, you make a selection as `selection = "around 5 resname ALA"`. For frame 1, let's assume this selection has 10 atoms, and for frame 50, it has 20 atoms. Do you expect RMSD(frame 1, frame 50) and RMSD(frame 50, frame 1) in the `DistanceMatrix` to be different since the selection is being updated dynamically due to the UpdatingAtomGroup? If this is what was implied here  https://github.com/MDAnalysis/mdanalysis/pull/3541/files/2799ad9cf2b9f12bf673a217985d781535f802a9#r831565393 , I believe it doesn't work this way. 
What can be done is package all these dynamic selections into one atomgroup and run our `DistanceMatrix` analysis on this AG (static selection).

Reply:
@IAlibay Is my understanding correct?

Reply:
@AnirG this is essentially the crux of what I was leading to with my original question on this, I was trying to make you think about the cases where a user passed in an UpdatingAtomgroup expecting that the there would be more / fewer atoms depending on the frames - would things work as expected?

Because of the way `metric` works, there is an expectation that the `i_ref` and `j_ref` arrays are of equal lengths (see https://github.com/MDAnalysis/mdanalysis/blob/009c467511210c442d15e4014f02e62e07dc9e85/package/MDAnalysis/analysis/diffusionmap.py#L185-L191), so as you observed above, this is unlikely to work.

It's fine in this case for the solution to just be "UpdatingAtomGroups are not accepted". Creating a complex solution involving the eventual generation of a static selection is probably not worth it.

The next steps here are then - what code changes are necessary here to avoid users passing UpdatinAtomGroups?

Reply:
As @lilyminium just reminded me, https://github.com/MDAnalysis/mdanalysis/blob/e6f0fe1289d150747befd1f1773db5657760f2c5/package/MDAnalysis/analysis/diffusionmap.py#L256 essentially means that all UpdatingAtomGroups will be squashed to AtomGroups on class initilisation. Probably all you need here is to have a check on the input to see if it's an UpdatingAtomGroup and warn users that it is being converted to a static AtomGroup based on the current frame selection.

Reply:
@IAlibay thanks for clarifying, I'll make the necessary changes.